packages/backend/src/services/SlackService/SlackService.ts

```diff
@@ -0,0 +1,513 @@
+        channels: slackUserId, // event.channel
+        token: context.userToken,
+        file: screenshot,
+        filename: `dashboard-screenshot-${imageId}.png`,
+        // thread_ts: event.message_ts, // Upload on thread
+    });
+    const publicImage = await client.files.sharedPublicURL({
+        file: fileUpload.file.id,
+        token: context.userToken,
+    });
+
+    const permalink = publicImage?.file?.permalink_public;
+    const permalinkParts = permalink.split('-');
+    const pubSecret = permalinkParts[permalinkParts.length - 1];
+    const imageUrl = `${publicImage?.file.url_private}?pub_secret=${pubSecret}`;
+    return imageUrl;
+};
+
+const saveScreenshot = async (
+    imageId: string,
+    url: string,
+    cookie: string,
+    lightdashPage: LightdashPage,
+): Promise<Buffer> => {
+    let browser;
+
+    try {
+        const browserWSEndpoint = `ws://${process.env.HEADLESS_BROWSER_HOST}:${process.env.HEADLESS_BROWSER_PORT}`;
+        browser = await puppeteer.connect({
+            browserWSEndpoint,
+        });
+
+        const page = await browser.newPage();
+
+        await page.setExtraHTTPHeaders({ cookie });
+
+        await page.setViewport({
+            width: 1400,
+            height: 768, // hardcoded
+        });
+
+        const blockedUrls = [
+            'headwayapp.co',
+            'rudderlabs.com',
+            'analytics.lightdash.com',
+            'cohere.so',
+            'intercom.io',
+        ];
+        await page.setRequestInterception(true);
+        page.on('request', (request: any) => {
+            const requestUrl = request.url();
+            if (blockedUrls.includes(requestUrl)) {
+                request.abort();
+                return;
+            }
+
+            request.continue();
+        });
+        await page.goto(url, {
+            timeout: 100000,
+            waitUntil: 'networkidle0',
+        });
+        const path = `/tmp/${imageId}.png`;
+
+        const selector =
+            lightdashPage === LightdashPage.DASHBOARD
+                ? '.react-grid-layout'
+                : `.echarts-for-react, [data-testid="visualization"]`; // Get .echarts-for-react, otherwise fallsback to data-testid (tables and bignumbers)
+        await page.waitForSelector(selector);
+        const element = await page.$(selector);
+
+        if (lightdashPage === LightdashPage.DASHBOARD) {
+            // Remove navbar from screenshot
+            await page.evaluate((sel: any) => {
+                // @ts-ignore
+                const elements = document.querySelectorAll(sel);
+                elements.forEach((el) => el.parentNode.removeChild(el));
+            }, '.bp4-navbar');
+        }
+
+        const imageBuffer = await element.screenshot({
+            path,
+        });
+
+        return imageBuffer;
+
+        // return path
+    } catch (e) {
+        console.error(`Unable to fetch screenshots from headless chromeo ${e}`);
```

Comment:
Would be better to use `Logger.error` instead of `console.error` to keep the logging with consistent styling.